ui/js/controllers/perf/alerts.js

```diff
@@ -506,6 +507,31 @@ perf.controller('AlertsCtrl', [
                     $scope.alertSummaries = _.union($scope.alertSummaries,
                                                     alertSummaries);
                 }
+
+                // Filling in downstream alertSummaries
+                _.forEach($scope.alertSummaries, function(alertSummary) {
+                    alertSummary.downstreamAlertSummaries = [];
+                    _.forEach(alertSummary.alerts, function(alert) {
+                        // filter for showing downstream alertSummaries
+                        if (alert.status === phAlertStatusMap.DOWNSTREAM.id &&
+                            alert.summary_id !== alertSummary.id ) {
+                            PhAlerts.getAlertSummary(alert.summary_id).then(
```

Comment:
Getting this every time is going to be super slow. We should just get the id in this step. You can just grab that from the `summary_id` property of `alert`. Since we just need the integer id, we could probably simplify this to something like (untested):

```js
alertSummary.downstreamSummaryIds = _.uniq(_.flatten(_.map(alertSummary.alerts, function(alert) {
  if (alert.status === phAlertStatusMap.DOWNSTREAM.id && alert.summary_id !== alertSummary.id ) {
      return alert.summary_id;
  } else {
      return [];
  }
})));
```

Check out the lodash documentation if you're unclear on what any of that means. :)